web/src/main/java/org/cbioportal/web/GenericAssayController.java

```diff
@@ -48,6 +51,25 @@
     @Autowired
     private GenericAssayService genericAssayService;
 
+    @RequestMapping(value = "/generic_assay_meta/{genericAssayType}", method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)	
```

Comment:
@dippindots @pvannierop What is the point of this endpoint?

Having said that, I suppose we don't need permissions on this because this data can be though of as gene information - freely available.

Reply:
@n1zea144 It's because the treatment validation will need to use this endpoint to get all the `treatment` entities from the database to validate the data (like `geneset` or `gene-panels`).https://github.com/cBioPortal/cbioportal/commit/cd57f5967e1b8c6280a65053a37a23f2ed3e38cb#diff-fee5ab3c72d7aa124ff0d2d3b199c68bR4614-R4615